### ISSUE TITLE
register projections are flagged as var-naming[non-string]

### ISSUE THREAD

**Issue by briantist** [ansible/ansible-lint#5093]
##### Summary

Register projections:
- https://forum.ansible.com/t/where-is-the-documentation-for-register-projections-and-task/45934/2
- https://github.com/ansible/ansible/pull/86241
- https://github.com/ansible/ansible-documentation/pull/3752
- https://ansible--3752.org.readthedocs.build/projects/ansible-core/3752/playbook_guide/playbooks_variables.html#registering-variables

When using register projections, for example:

```yaml
- name: My task
  register:
    reg_var1: _task.result
  ansible.builtin.debug:
    msg: hello
```

The following failure is the result:

```
var-naming[non-string]: Variables names must be strings. (register: {'reg_var1': '_task.result', '__file__': PosixPath('playbooks/test.yml')})
```

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console
ansible-lint 26.6.0 using ansible-core:2.21.0 ansible-compat:26.3.0 ruamel-yaml:0.19.1 ruamel-yaml-clib:None
```

- ansible installation method: pip

##### STEPS TO REPRODUCE

See summary.

##### Desired Behavior

Lint passes.

##### Actual Behavior

See summary.

**Comment by tigattack** [comment]
Some movement on this would be greatly appreciated; it's quite frustrating for a feature that went GA ~2 months ago to be the result of persistent lint failures.
